source/read/change-streams.txt

```diff
@@ -0,0 +1,261 @@
+.. _php-change-streams:
+
+====================
+Monitor Data Changes
+====================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: watch, code example
+
+Overview
+--------
+
+In this guide, you can learn how to use a **change stream** to monitor real-time
+changes to your data. A change stream is a {+mdb-server+} feature that
+allows your application to subscribe to data changes on a collection, database,
+or deployment.
+
+When using the {+php-library+}, you can create an instance of ``MongoDB\ChangeStream`` to
+monitor data changes, such as updates, insertions, and deletions.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To access this collection
+from your PHP application, instantiate a ``MongoDB\Client`` that connects to an Atlas cluster
+and assign the following value to your ``collection`` variable:
```

Comment:
`$client->sample_restaurants->restaurants` constructs a new collection, so it'd be more accurate to say something like "select a collection to assign to the ``collection`` variable" or "instantiate your ``collection`` variable".

Also, since you're referring to a variable by name here, would it make sense to use ``$collection``? That would be consistent with how use a "()" suffix when referring to methods like `watch()` below.

Reply:
Fixed!

Reply:
Changed to `$collection`! Do you mean that `$client->sample_restaurants->restaurants` creates a new `restaurants` collection rather than accessing the `restaurants` sample collection? From running the code, it does query the sample data

Reply:
My objection was with the word "value" in "assign the following value" because `$client->sample_restaurants->restaurants` is not actually a value even though it looks like we're accessing PHP properties.

The code here is using the [`Database::__get()`](https://github.com/mongodb/mongo-php-library/blob/1.19.1/src/Database.php#L163) method, which delegates to [`Database::selectCollection()`](https://github.com/mongodb/mongo-php-library/blob/1.19.1/src/Database.php#L559). That method constructs a new `MongoDB\Collection` object each time it is called. To me, "value" implies that we're assigning some fixed value.

If it's kosher to write "assign the following to your `$collection` variable:" I'd be in favor of that. But if "value" needs to stay we can resolve this thread without further changes.

Note: In the PHPLIB docs I've historically used the `Client::selectDatabase()`, `Client::selectCollection()`, and `Database::selectCollection()` methods; however, I understand that you might want to use the shorthand syntax for consistency with other driver/shell docs.

> Do you mean that $client->sample_restaurants->restaurants creates a new restaurants collection rather than accessing the restaurants sample collection?

Creation of the `MongoDB\Collection` object has no impact on the server. We're just initializing the object with strings to identify the database and collection (i.e. namespace). The driver will only create a collection on the server if we explicitly invoke the [`create` command](https://www.mongodb.com/docs/manual/reference/command/create/) or write to a collection that doesn't exist (implicit creation).